pdns/slavecommunicator.cc

```diff
@@ -745,15 +745,7 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
     TSIGRecordContent trc;
     DNSName tsigkeyname;
     dp.getTSIGDetails(&trc, &tsigkeyname);
-    int res;
-    res=P->trySuperMasterSynchronous(&dp, tsigkeyname);
-    if(res>=0) {
-      DNSPacket *r=dp.replyPacket();
-      r->setRcode(res);
-      r->setOpcode(Opcode::Notify);
-      N->send(r);
-      delete r;
-    }
+    P->trySuperMasterSynchronous(&dp, tsigkeyname); // FIXME could use some error loging
```

Comment:
was there a specific reason for removing the error packet generation? or can it no longer fail?

Reply:
this was responsible for the spurious answer (log is in the commit message). After removal of this piece of code the error is gone. There is no way to let the master know at this point it is failing.
A nice error in the log is all we can do here.